workspaces/ui-v2/src/lib/shape-diff-dsl-rust.ts

```diff
@@ -135,7 +137,10 @@ export class Actual {
     private shapeTrail: IShapeTrail,
     public jsonTrail: IJsonTrail
   ) {
-    this.trailAffordances = learnedTrails.affordances;
+    this.trailAffordances = learnedTrails.affordances.filter((i) => {
```

Comment:
@JaapRood @LouManglass @devdoshi this is why the choices presented for some of the diffs were incorrect. Optic was not filtering these in the right spot. You want the `trailAffordances` filtered when you are presenting options to update it, but you want all of them when you're building the nested shapes.